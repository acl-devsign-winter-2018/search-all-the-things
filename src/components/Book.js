import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Article extends Component {

  static propTypes = {
    article: PropTypes.object.isRequired
  };

  render() {
    const { title, authors, description, publishedDate, url } = this.props.article.volumeInfo;
    const imageLinks = this.props.article.volumeInfo.imageLinks;

    return (
      <li>
        <a href={url} target="_blank">
          <h2>{title} by {authors} on {publishedDate}</h2>
          <p>{description}</p>
          {imageLinks ? <img src={imageLinks.thumbnail}/> : <div>no image</div>}
        </a>
      </li>
    );
  }
}