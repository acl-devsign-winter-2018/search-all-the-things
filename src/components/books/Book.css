li {
  display: block;
  align-items: center;
}

a {
  color: black;
  text-decoration: none;
}

a:active {
  color: teal;
}