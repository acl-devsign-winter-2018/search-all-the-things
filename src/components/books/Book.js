import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Book.css';

export default class Book extends Component {

  static propTypes = {
    book: PropTypes.object.isRequired,
  };
  
  render() {

    const { selfLink } = this.props.book;
    const { title, publishedDate, description } = this.props.book.volumeInfo;
    
    const thumbnail = (this.props.book.volumeInfo.imageLinks) ? this.props.book.volumeInfo.imageLinks.thumbnail : 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTc5ewfk-9xrAwUt2clcSPwklh4-lfK2qDvROMNcUsSy3VmKGI_';

    const authors = (this.props.book.volumeInfo.authors) ? this.props.book.volumeInfo.authors : ['No Authors Listed'];
    
    function handleAuthors(authors) {
      if(authors.length > 1){
        return authors.join(', ');
      } 
      else return authors;
    }

    return (
      <li>
        <a href={selfLink} target="_blank"> 
          <h2>{title} by {handleAuthors(authors)}</h2>
          <img src={thumbnail}/> 
          <p className="published"> Published {publishedDate} </p>
          <p>{description}</p> 
        </a>
      </li>
    );
  }
}