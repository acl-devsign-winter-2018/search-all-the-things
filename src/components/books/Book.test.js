import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Book from './Book';


describe('Book', () => {
  it('renders with props', () => {
    
    const item = 
      {
        kind: 'books#volume',
        id: 'UWy3AQAACAAJ',
        etag: 'mWuxVpRF0Hw',
        selfLink: 'https://www.googleapis.com/books/v1/volumes/UWy3AQAACAAJ',
        volumeInfo: {
          title: 'Coming to the Fire',
          subtitle: 'Exploring Our Shared History with Dogs, Cats, Cows, and Horses',
          authors: [
            'Gavin Ehringer'
          ],
          publishedDate: '2017-12-05',
          description: 'A thought-provoking and surprising book that explores the ever-evolving relationship between humans and domesticated animals.',
        },
        imageLinks: {
          thumbnail: 'http://books.google.com/books/content?id=UWy3AQAACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api'
        },
      };

    const wrapper = shallow(<Book book={item}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();

  });
});