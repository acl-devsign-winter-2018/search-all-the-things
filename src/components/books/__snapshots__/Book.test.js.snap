// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book renders with props 1`] = `
<li>
  <a
    href="https://www.googleapis.com/books/v1/volumes/UWy3AQAACAAJ"
    target="_blank"
  >
    <h2>
      Coming to the Fire
       by 
      Gavin Ehringer
    </h2>
    <img
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTc5ewfk-9xrAwUt2clcSPwklh4-lfK2qDvROMNcUsSy3VmKGI_"
    />
    <p
      className="published"
    >
       Published 
      2017-12-05
       
    </p>
    <p>
      A thought-provoking and surprising book that explores the ever-evolving relationship between humans and domesticated animals.
    </p>
  </a>
</li>
`;
